packages/design-system/src/components/Table/Table.jsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Alert from '../Alert/Alert';
+import PropTypes from 'prop-types';
+import TableCaption from './TableCaption';
+import classNames from 'classnames';
+
+function debounce(fn, ms) {
+  let timer;
+  return () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+      timer = null;
+      fn.apply(this, arguments);
+    }, ms);
+  };
+}
+
+export const Table = ({
+  className,
+  stacked,
+  striped,
+  scrollTable,
+  scrollableCaption,
+  children,
+  ...attributeOptions
+}) => {
+  const container = useRef(null);
+  const captionID = useRef('caption-' + Math.random().toString(36).substr(2, 9));
+  const [tabIndex, setTabIndex] = useState(null);
+
+  useEffect(() => {
+    if (scrollTable) {
+      const { scrollWidth, clientWidth } = container.current;
+      const scrollable = scrollWidth > clientWidth;
+      setTabIndex(scrollable ? '0' : null);
```

Comment:
Is this still necessary? I'm not entire sure what this function does but I thought we don't need to set tab index anymore

Reply:
Not exactly sure.
From https://inclusive-components.design/data-tables/ - it talks about horizontal scroll needs tabIndex for keyboard support. Then to give screen reader context of the tabIndex focus, one way is to tie this `TabIndex=='0' `to the Caption via `captionID`.
@davidakennedy - would you be able to help confirm which way to go for this?